tests/data/cases/fmtskip10.py

```diff
@@ -0,0 +1 @@
+def foo(): return "mock"  # fmt: skip
```

Comment:
```suggestion
def foo(): return "mock"  # fmt: skip
if True: print("yay")  # fmt: skip
for i in range(10): print(i)  # fmt: skip
```
Looks to also fix #3682 so this adds two examples from that as test cases. The other issues not fixed by this PR in #3682 look to have been fixed by something in the 03-15-2024 release, probably #4248, so it should be good to add to the close-on-merge list for this PR along with #4535
(To do that, edit the opening description to include
```
Fixes #3682
Fixes #4535
```
)

Reply:
Thanks for the review.

I added the suggested tests + some extra ones.
I also mentioned #3682 on the PR description as suggested.